src/Integration.Vsix.UnitTests/SonarLintTagger/TaggerProviderTests.cs

```diff
@@ -324,6 +369,13 @@ private void SetupDetectedLanguages(string fileName, IContentType bufferContentT
                 .Returns(detectedLanguages);
         }
 
+        private SingletonDisposableTaggerManager<IErrorTag> CheckPresenceOfSingletonManagerInPropertyCollection(ITextBuffer buffer, bool shouldExist)
```

Comment:
I think the `shouldExist` makes the usage hard to understand, as in it's hard to understand this flag does
`var doc1Manager = CheckPresenceOfSingletonManagerInPropertyCollection(doc1.TextBuffer, true);` 

Is it important to check what TryGetProperty returns, instead of just checking the outcome of the propertyValue? I think it will be more readable to just add a null assertion after the call
```
var doc1Manager = CheckPresenceOfSingletonManagerInPropertyCollection(doc1.TextBuffer);
doc1Manager.Should().BeNotNull();
```